CMakeLists.txt

```diff
@@ -41,18 +41,13 @@ find_package(Z3 CONFIG REQUIRED)
 
 # LLVM
 find_package(LLVM CONFIG REQUIRED)
-llvm_map_components_to_libnames(llvm_libs
-  support core irreader
-  bitreader bitwriter
-  passes asmprinter
-  aarch64codegen aarch64asmparser
-  armcodegen armasmparser
-  interpreter mcjit
-  nvptxdesc
-  x86codegen x86asmparser
-  sparccodegen sparcasmparser
-  webassemblydesc)
-message(STATUS "LLVM Libraries: ${llvm_libs}")
+
+# https://github.com/JonathanSalwan/Triton/issues/1082#issuecomment-1030826696
+if(LLVM_LINK_LLVM_DYLIB)
+  set(llvm_libs LLVM)
+else()
+  set(llvm_libs ${LLVM_AVAILABLE_LIBS})
```

Comment:
Still want `llvm_map_components_to_libnames` here, otherwise it starts to pull in things like `LibEdit`. In the past we've also faced linking issues with `LLVM_AVAILABLE_LIBS` having some things out-of-order.

Reply:
Yeah that's true, I had to disable `LibEdit` on the LLVM command line. I'll try to reproduce the linker error then and see if I can find the component. Generally it shouldn't be an issue to link to all LLVM libraries, unused things are ignored. I've been using this since LLVM 10 without issues related to out-of-order linking.